app/forms/reactivate_profile_form.rb

```diff
@@ -0,0 +1,69 @@
+class ReactivateProfileForm
+  include ActiveModel::Model
+
+  validates :recovery_code, :password, presence: true
+  validate :validate_password_reset_profile
+  validate :validate_password
+  validate :validate_recovery_code
+
+  attr_accessor :recovery_code, :password
+  attr_reader :user
+
+  def initialize(user, attrs = {})
+    @user = user
+    super attrs
+  end
+
+  def submit(flash)
+    if valid?
+      flash[:recovery_code] = reencrypt_pii
+      true
+    else
+      clear_fields
+      false
+    end
+  end
+
+  protected
+
+  def reencrypt_pii
+    profile = user.password_reset_profile
+    pii = profile.recover_pii(recovery_code)
+    user_access_key = user.unlock_user_access_key(password)
+    recovery_code = profile.encrypt_pii(user_access_key, pii)
+    profile.deactivation_reason = :not_deactivated
+    profile.save!
+    recovery_code
+  end
+
+  def validate_password_reset_profile
+    errors.add :base, :no_password_reset_profile unless user.password_reset_profile
+  end
+
+  # rubocop:disable Style/GuardClause
```

Comment:
well it puts me in a pickle for the recovery_code one:

Rubocop wants this to be a guard clause:

```
  def validate_recovery_code
    if recovery_code.present? && !valid_recovery_code?
      errors.add :recovery_code, :recovery_code_incorrect
    end
  end
```

which is this:

```
  def validate_recovery_code
    errors.add :recovery_code, :recovery_code_incorrect if recovery_code.present? && !valid_recovery_code?
  end
```

which then rubocop gets mad at for a line length violation.

and I know we *can* move the condition to the next line, but I really really really dislike that approach, so this was the least of all evils to me.

Would you prefer the if condition on the next line, or just disable line length?